### ISSUE TITLE
@sap/hana-client crash on Windows

### ISSUE THREAD

**Issue by SEIDOR-JCHEN** [oven-sh/bun#12916]
### How can we reproduce the crash?

``` javascript
var hana = require('@sap/hana-client');

var conn = hana.createConnection(); //Crash here

var conn_params = {
  serverNode  : 'myserver:30015',
  uid         : 'system',
  pwd         : 'Password123'
};

conn.connect(conn_params, function(err) {
 //...
});
```

I cannot use the @sap/hana-client driver to connect to the HANA database. It crashes when creating the connection. I looked into issue #5139 but did not get any results; I think that issue solves problems for Linux and macOS. I found some relation to issue #4290, but I'm not sure.

### Relevant log output

```shell
Bun v1.1.21 (70ca2b76) Windows x64
CPU: sse42 avx avx2
Args: "C:\Users\llabsap2.LLABERIA\.bun\bin\bun.exe" "run" "--hot" "--inspect" "src/index.ts"
Features: jsc Bun.stderr Bun.stdout dotenv(4) http_server tsconfig_paths(2) tsconfig(10) 
Builtins: "bun:main" "node:assert" "node:buffer" "node:crypto" "node:events" "node:fs" "node:http" "node:https" "node:net" "node:os" "node:path" "node:stream" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" 
Elapsed: 1451ms | User: 937ms | Sys: 546ms
RSS: 0.24GB | Peak: 0.24GB | Commit: 0.31GB | Faults: 58121

panic(main thread): Segmentation fault at address 0x5
oh no: Bun has crashed. This indicates a bug in Bun, not your code.
```

### Stack Trace (bun.report)

Bun v1.1.21 ([`70ca2b7`](<https://github.com/oven-sh/bun/tree/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b>)) on windows x86_64 [RunCommand]

Segmentation fault at address 0x00000005

- [`JSSQLStatement.cpp:2078`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L2078>): `jsSQLStatementExecuteStatementFunctionGetWithoutTypeChecking`
- ??? at `0x310b2` in hana-client.node
- ??? at `0x3125a` in hana-client.node
- ??? at `0xac03` in hana-client.node
- ??? at `0xcf49` in hana-client.node
- [`napi.cpp:269`](<https://github.com/oven-sh/bun/blob/70ca2b76c3d33f7ceeea43a702b396ab8ffce25b/src/bun.js/bindings/napi.cpp#L269>): `NAPICallFrame::dataPtr`
- *2 unknown/js code*
- `llint_entry`

**Comment by Jarred-Sumner** [comment]
What is hana-client?

**Comment by Jarred-Sumner** [comment]
Strange the top of the stack trace is sqlite.

**Comment by nektro** [comment]
I no longer experience a crash but using your invocation and then saving the file i get

```
TypeError: symbol 'napi_register_module_v1' not found in native module. Is this a Node API (napi) module?
      at C:\Users\dave\.bun\install\cache\@sap\hana-client@2.22.29@@@1\lib\index.js:758:14
      at require (1:11)
      at C:\test\foo\index.mjs:1:5
```

**Comment by SEIDOR-JCHEN** [comment]
Hi,

I believe the problem has been resolved after version 1.1.30.

I'm not sure if it's an ECM or CJS issue causing your error, but I'm using the following code, and it works for me:

```
import { createConnection } from '@sap/hana-client';

const client = createConnection({
  host: "...",
  port: "...",
  uid: "...",
  pwd: "...",
});

client.connect(function (err) {
  if (err) {
    reject(err);
  } else {
    console.log('Connection to the HANA database established');
  }
});
```
